### ISSUE TITLE
Prevent certain songs from getting queued

### ISSUE THREAD

**Issue by ghost** [vfsfitvnm/ViMusic#277]
### Feature description

I would like to be able to stop certain songs (and or artists) from getting auto-qued (or have them auto-removed).

Maybe a button on the play/ artist page with confirmation and to undo it you can still find it in the search.

### Why do you want this feature?

Im annoyed of a few certain song which are played over and over even tho i dont like them (and often dont like the author in general).

### Tags (added by maintainer)
blacklist, denylist

**Comment by vfsfitvnm** [comment]
It may be useful, but (from my own experience) I wouldn't have the patience to open the app and block a song: I'd rather skip it as it requires less effort.

However, I could add a very simple algorithm to automatically determine whether to block a song (it just crossed my mind now, I don't know if it's robust enough). The blacklist would be based on 4 levels.

1 - Skipped songs you never listened to before with less than 1 minute of current playback duration (you start listening to them, but you don't like them)
2 - Skipped songs from level 1 with less than 30 seconds of current playback duration (you recognize the songs you don't like)
3 - Skipped songs from level 2 with less than 15 seconds of current playback duration (you recognize that song you don't like even faster)

When a song reaches the level 3, then it gets blocked and it will removed from the radio queue and youtube playlists.

**Comment by Jak2k** [comment]
I would prefer to do it manualy.

**Comment by BluePixel4k** [comment]
> I would prefer to do it manualy.

Me too.

Maybe you can add a "dislike" button for that? Like Spotify doing it for auto generated playlists (⊖).
<img src="https://user-images.githubusercontent.com/77979781/188719661-a962d2dd-7b5b-42b5-b7f6-ca0079f32869.png" width="300" />
<img src="https://user-images.githubusercontent.com/77979781/188719656-2864dd00-4fd1-4fe0-ba02-f0d7ed0cf6bf.png" width="300" />

And if you dislike maybe three times a song from an artist, you can be asked if you want to hide all songs from him.

**Comment by Khenxy** [comment]
> 



> It may be useful, but (from my own experience) I wouldn't have the patience to open the app and block a song: I'd rather skip it as it requires less effort.
> 
> However, I could add a very simple algorithm to automatically determine whether to block a song (it just crossed my mind now, I don't know if it's robust enough). The blacklist would be based on 3 levels.
> 
> 1 - Skipped songs you never listened to before with less than 1 minute of current playback duration (you start listening to them, but you don't like them) 2 - Skipped songs from level 1 with less than 30 seconds of current playback duration (you recognize the songs you don't like, so you skip it earlier) 3 - Skipped songs from level 2 with less than 15 seconds of current playback duration (you recognize that song you don't like even faster)
> 
> When a song reaches the level 3, then it gets blocked and it will removed from the radio queue and youtube playlists.




That's would be a addition to the app. Also could you please add a feature that prevents incompletely cached songs from playing when you're playing offline because the stop midway and interrupt the whole Playlist, it's inconvenient to always pick up your phone to next the songs when it stops and most times I'm not even close to my phone.
Thanks.

**Comment by vfsfitvnm** [comment]
@Khenxy If you are offline, only play songs from the "offline" 'playlist

**Comment by Khenxy** [comment]
Yeah I play only from the offline Playlist but I still experience this cause some songs are just downloaded halfway especially songs I don't allow to play till the end when I was online.  
That's an example in the image I attached 

![Screenshot_20221003-154335199](https://user-images.githubusercontent.com/114582293/193606264-eaddaf8c-b823-46bb-bd2e-ee560c66ad01.png)

**Comment by vfsfitvnm** [comment]
The offline playlist should show songs that are 100% cached

**Comment by Khenxy** [comment]
Exactly but it doesn't at least not in the version I have.
